src/main/java/io/nats/client/impl/NatsJetStreamManagement.java

```diff
@@ -377,32 +373,30 @@ public LinkedBlockingQueue<MessageInfo> queueMessageBatch(String streamName, Mes
     @Override
     public void requestMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest, MessageInfoHandler handler) throws IOException, JetStreamApiException {
         validateMessageBatchGetRequest(streamName, messageBatchGetRequest);
-        _requestMessageBatch(streamName, messageBatchGetRequest, handler);
+        _requestMessageBatch(streamName, messageBatchGetRequest, true, handler);
     }
 
-    public void _requestMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest, MessageInfoHandler handler) {
+    public void _requestMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest, boolean sendEod, MessageInfoHandler handler) {
         Subscription sub = null;
+
         try {
             String replyTo = conn.createInbox();
             sub = conn.subscribe(replyTo);
 
             String requestSubject = prependPrefix(String.format(JSAPI_DIRECT_GET, streamName));
             conn.publish(requestSubject, replyTo, messageBatchGetRequest.serialize());
 
-            long maxTimeMillis = getTimeout().toMillis();
-            long timeLeft = maxTimeMillis;
-            long start = System.currentTimeMillis();
             while (true) {
-                Message msg = sub.nextMessage(timeLeft);
+                Message msg = sub.nextMessage(getTimeout());
```

Comment:
Since there is a sentinel, one timeout is sufficient. We don't need all the timeout calculations.